framework/PostProcessors/FTImporter.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on Dec 21, 2017
+
+@author: mandd
+
+"""
+
+from __future__ import division, print_function , unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default', DeprecationWarning)
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+import xml.etree.ElementTree as ET
+import copy
+import itertools
+from collections import OrderedDict
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from .PostProcessor import PostProcessor
+from utils import InputData
+from utils import xmlUtils as xmlU
+from utils import utils
+from .FTstructure import FTstructure
+import Files
+import Runners
+#Internal Modules End-----------------------------------------------------------
+
+class FTImporter(PostProcessor):
+  """
+    This is the base class of the postprocessor that imports Fault-Trees (FTs) into RAVEN as a PointSet
+  """
+  def __init__(self, messageHandler):
+    """
+      Constructor
+      @ In, messageHandler, MessageHandler, message handler object
+      @ Out, None
+    """
+    PostProcessor.__init__(self, messageHandler)
+    self.printTag = 'POSTPROCESSOR FT IMPORTER'
+    self.FTFormat = None
+    self.allowedFormats = ['OpenPSA']
+
+    self.topEventID = None
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(FTImporter, cls).getInputSpecification()
+    inputSpecification.addSub(InputData.parameterInputFactory("fileFormat", contentType=InputData.StringType))
```

Comment:
use EnumType.... where you can add all the "allowed" formats and avoid to check for them later on

Reply:
added.